src/PrestaShopBundle/Form/Admin/Improve/International/Locations/StateType.php

```diff
@@ -0,0 +1,174 @@
+use Symfony\Component\Translation\TranslatorInterface;
+use Symfony\Component\Validator\Constraints\Length;
+use Symfony\Component\Validator\Constraints\NotBlank;
+
+class StateType extends AbstractType
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+    /**
+     * @var ConfigurableFormChoiceProviderInterface
+     */
+    private $countriesChoiceProvider;
+    /**
+     * @var ConfigurableFormChoiceProviderInterface
+     */
+    private $zonesChoiceProvider;
+
+    /**
+     * StateType constructor.
+     *
+     * @param TranslatorInterface $translator
+     * @param ConfigurableFormChoiceProviderInterface $countriesChoiceProvider
+     * @param ConfigurableFormChoiceProviderInterface $zonesChoiceProvider
+     */
+    public function __construct(
+        TranslatorInterface $translator,
+        ConfigurableFormChoiceProviderInterface $countriesChoiceProvider,
+        ConfigurableFormChoiceProviderInterface $zonesChoiceProvider
+    ) {
+        $this->translator = $translator;
+        $this->countriesChoiceProvider = $countriesChoiceProvider;
+        $this->zonesChoiceProvider = $zonesChoiceProvider;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $stateIdValue = isset($builder->getData()['id_state']) && $builder->getData()['id_state'] instanceof StateId ?
+            $builder->getData()['id_state']->getValue() :
+            null;
+
+        $builder
+            ->add('name', TextType::class, [
+                'required' => true,
+                'help' => $this->translator->trans(
+                    'Provide the state name to be displayed in addresses and on invoices.',
+                    [],
+                    'Admin.International.Help'
+                ),
+                'label' => $this->translator->trans('Name', [], 'Admin.Global'),
+                'constraints' => [
+                    new NotBlank([
+                        'message' => $this->translator->trans(
+                            'This field cannot be empty.', [], 'Admin.Notifications.Error'
+                        ),
+                    ]),
+                    new Length([
+                        'max' => 64,
+                        'maxMessage' => $this->translator->trans(
+                            'This field cannot be longer than %limit% characters.',
+                            ['%limit%' => 64],
+                            'Admin.Notifications.Error'
+                        ),
+                    ]),
+                    new TypedRegex([
+                        'type' => 'generic_name',
+                    ]),
+                ],
+            ])
+            ->add('iso_code', TextType::class, [
+                'label' => $this->translator->trans('ISO code', [], 'Admin.International.Feature'),
+                'help' => $this->translator->trans('1 to 4 letter ISO code', [], 'Admin.International.Help')
```

Comment:
```suggestion
                'help' => $this->translator->trans('1 to 4 letter ISO code.', [], 'Admin.International.Help')
```

I had missed this one. Let's use the existing string :-)

Reply:
Fixed